packages/pwa-kit-react-sdk/src/ssr/server/react-rendering.js

```diff
@@ -134,19 +134,29 @@ const initAppState = async ({App, component, match, route, req, res, location})
  * @return {Promise}
  */
 export const render = async (req, res) => {
+    const [pathname, search] = req.originalUrl.split('?')
+    const location = {
+        pathname,
+        search: search ? `?${search}` : ''
+    }
+
+    // Step 0 - Get the localization props and put it in our locals
+    const intlConfig = await AppConfig.getIntlConfig({
+        req,
+        location
+    })
+
+    // TODO: Think about if this is a good place to put the locale, or just pass it
+    // into `restore` as an additional argument since it's essentially in the api now.
+    res.locals.locale = intlConfig?.locale
```

Comment:
this is a question worth some discussion. I don't have a stance on whether we should embed intl into the `locals` or baking `intl` into the api signatures. I'll think about this tonight and get back to this thread..

But i have a suggestion to use an express middleware to handle localization... since `render` function is just a normal express handler, we could have a middleware like:

```js
export const intlMiddleware = (req, res, next) => {
  const [pathname, search] = req.originalUrl.split('?')
    const location = {
        pathname,
        search: search ? `?${search}` : ''
    }

    // Step 0 - Get the localization props and put it in our locals
    const intlConfig = await AppConfig.getIntlConfig({
        req,
        location
    })

    // TODO: Think about if this is a good place to put the locale, or just pass it
    // into `restore` as an additional argument since it's essentially in the api now.
    res.locals.locale = intlConfig?.locale
    next()
}
```

and `Boolean(res.locals.locale)` becomes a feature flag that the rendering pipeline could use to enable/disable intl.

Then on the user land, user can enable intl by using this middleware like this:

```js
// in packages/pwa/app/ssr.js


import path from 'path'
import {createApp} from 'pwa-kit-react-sdk/ssr/server/express'
import {render} from 'pwa-kit-react-sdk/ssr/server/react-rendering'
import intlMiddleware from 'sdk'

const app = createApp(SSR_CONFIG)

app.use('/*', intlMiddleware) // this is where the developers interacts with our SDK to enable intl.
app.get('/*', render)

export const get = createHandler(app)
```

Reply:
We could take the middleware even further by making it an entry point for accepting the `intlConfig`.

```
export const intlMiddleware = (config) => {
  return (req, res, next) => {
    // we now have access to the configs
 
    next()
  }
}

// in ssr.js
app.use('/*', intlMiddleware(intlConfig))
```